### PR TITLE
added a state to calculator

### PR DESCRIPTION
**Pull request by SpaYco**
in this branch, I added the state to the `App` component to update the display through the `buttons`.

the [Live Demo](https://spayco-react-calculator.herokuapp.com/) is updated!

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -1,19 +1,28 @@
+/* eslint-disable react/button-has-type */
```
Comment (Forison) [inline src/components/Button.js]: * [x] Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))
- Do not make change to linter configs else you miss the main learning objective in relation to the rule.

File: src/components/Button.js [hunk 2]
```diff
@@ -1,19 +1,28 @@
...
 import React from 'react';
 import PropTypes from 'prop-types';
 
 // eslint-disable-next-line react/prefer-stateless-function
```
Comment (Forison) [inline src/components/Button.js]: - Do not make any changes in config files

File: src/components/ButtonPanel.js [hunk 3]
```diff
@@ -15,54 +16,60 @@ const rowStyles = {
 // eslint-disable-next-line  react/prefer-stateless-function
```
Comment (Forison) [inline src/components/ButtonPanel.js]: - Do not make any changes in config files

**Review by Forison — CHANGES_REQUESTED** [review]
## Status:: Change required❌
# 🙋‍♂️ @Spayco 
#### You have done a good job on this one so far, here are a few issues we have to address to get the project approved🤗.
* [x] The project has a descriptive README.md ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- Kindly make sure the image in the readme is visible to visitors. (Kindly check the image path).
* [x] Kindly address the issue with the implementation of `%`, currently it is not functioning as required.
- onClick `50` followed by click `%` the result should be 0.5.
- onClick `50` followed by `-`, `30`click `%` the result should be 0.2.
* [x] display field overflows when the digit is long, kindly address it. 
overflo
- Check my comment below👇.
### That is it, kindly make the change and submit a re-review👨‍🔧.
#### Happy coding👨‍🔧
Addo Forison

**Review by bellom — APPROVED** [review]
## Hi @Spayco  :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**